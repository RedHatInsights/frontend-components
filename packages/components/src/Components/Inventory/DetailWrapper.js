import React, { Suspense } from 'react';
import PropTypes from 'prop-types';
const AsyncInventory = React.lazy(() => import('./AsyncInventory'));
import { ScalprumComponent } from '@scalprum/react-core';
import { useHistory } from 'react-router-dom';
import { useStore } from 'react-redux';

const DetailWrapper = ({ fallback, ...props }) => {
    const history = useHistory();
    const store = useStore();
    return <ScalprumComponent
        history={history}
        store={store}
        fallback={fallback}
        appName="chrome"
        module="./DetailWrapper"
        scope="chrome"
        ErrorComponent={() => <Suspense fallback={fallback}>
            <AsyncInventory
                component="DetailWrapper"
                {...props}
                fallback={fallback} />
        </Suspense>}
        {...props}
    />;
};

DetailWrapper.propTypes = {
    fallback: PropTypes.any
};

export default React.forwardRef((props, ref) => <DetailWrapper {...props} ref={ref} />);
