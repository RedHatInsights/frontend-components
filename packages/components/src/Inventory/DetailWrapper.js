import React, { Suspense } from 'react';
import PropTypes from 'prop-types';
import { ScalprumComponent } from '@scalprum/react-core';
import { useStore } from 'react-redux';
import { Bullseye } from '@patternfly/react-core/dist/dynamic/layouts/Bullseye';
import { Spinner } from '@patternfly/react-core/dist/dynamic/components/Spinner';
import InventoryLoadError from './InventoryLoadError';
import classNames from 'classnames';
import WithHistory from './WithHistory';

const BaseDetailWrapper = (props) => {
  const store = useStore();
  const Cmp = props.component;
  return (
    <Cmp className={classNames(props.className, 'inventory')}>
      <Suspense fallback={props.fallback}>
        <ScalprumComponent
          history={props.history}
          store={store}
          appName="inventory"
          module="./DetailWrapper"
          scope="inventory"
          ErrorComponent={<InventoryLoadError component="InventoryDetailHead" {...props} />}
          ref={props.innerRef}
          {...props}
        />
      </Suspense>
    </Cmp>
  );
};

BaseDetailWrapper.propTypes = {
  fallback: PropTypes.node,
  innerRef: PropTypes.object,
  component: PropTypes.string,
  className: PropTypes.string,
  history: PropTypes.object,
};

/**
 * Inventory sub component.
 *
 * This component wraps entire system detail in order to show loading state and drawer (if enabled).
 */
const DetailWrapper = React.forwardRef((props, ref) => <BaseDetailWrapper innerRef={ref} {...props} />);

DetailWrapper.propTypes = {
  /** React Suspense fallback component. <a href="https://reactjs.org/docs/code-splitting.html#reactlazy" target="_blank">Learn more</a>. */
  fallback: PropTypes.node,
  /** Optional wrapper component */
  component: PropTypes.string,
  /** Optional classname applied to wrapper component */
  className: PropTypes.string,
};

DetailWrapper.defaultProps = {
  fallback: (
    <Bullseye className="pf-v6-u-p-lg">
      <Spinner size="xl" />
    </Bullseye>
  ),
  component: 'section',
};

const CompatiblityWrapper = (props, ref) => <WithHistory innerRef={ref} Component={DetailWrapper} {...props} />;

export default React.forwardRef(CompatiblityWrapper);
