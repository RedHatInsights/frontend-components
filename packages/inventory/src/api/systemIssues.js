import instance from '@redhat-cloud-services/frontend-components-utilities/files/interceptors';
export const systemIssuesInstance = instance;

export const cves = async (systemId) => {
    try {
        const [ low, moderate, important, critical ] = await Promise.all([ 2, 4, 5, 7 ].map((impact) => (
            instance.get(
                `/api/vulnerability//v1/systems/${systemId}/cves?page=1&page_size=1&impact=${impact}`
            )
        )));
        return { low, moderate, important, critical };
    } catch (_e) {
        return {};
    }
};

export const patch = async (systemId) => {
    try {
        return await instance.get(`/api/patch/v1/systems/${systemId}`);
    } catch (_e) {
        return {};
    }
};

export const advisor = async (systemId) => {
    try {
        return await instance.get(`/api/insights/v1/system/${systemId}/reports/`);
    } catch (_e) {
        return {};
    }
};

export const compliance = async (systemId) => {
    const query = `query System($systemId: String!) {
        system(id: $systemId) {
            id
            name
            profiles {
                id
                name
                refId
                compliant
                rulesFailed
                rulesPassed
                lastScanned
                score
                __typename
            }
            __typename
        }
    }`;
    try {
        return await instance.post('/api/compliance/graphql', {
            operationName: 'System',
            query,
            variables: {
                systemId
            }
        });
    } catch (_e) {
        return {};
    }
};
