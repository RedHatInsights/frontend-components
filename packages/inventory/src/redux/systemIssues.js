/* eslint-disable camelcase */
import { SYSTEM_ISSUE_TYPES } from './action-types';

export function advisor(state, { payload }) {
    return {
        ...state,
        systemIssues: {
            ...state?.systemIssues,
            advisor: {
                isLoaded: true,
                criticalCount: payload?.filter(item => item?.total_risk >= 4)
            }
        }
    };
}

export function cve(state, { payload }) {
    return {
        ...state,
        systemIssues: {
            ...state?.systemIssues,
            cve: {
                isLoaded: true,
                critical: {
                    count: payload?.critical?.meta?.total_items || 0
                },
                moderate: {
                    count: payload?.moderate?.meta?.total_items || 0
                },
                important: {
                    count: payload?.important?.meta?.total_items || 0
                },
                low: {
                    count: payload?.low?.meta?.total_items || 0
                }
            }
        }
    };
}

export function patch(state, { payload }) {
    return {
        ...state,
        systemIssues: {
            ...state?.systemIssues,
            patch: {
                isLoaded: true,
                bug: {
                    count: payload?.data?.attributes?.rhba_count || 0
                },
                enhancement: {
                    count: payload?.data?.attributes?.rhea_count || 0
                },
                security: {
                    count: payload?.data?.attributes?.rhsa_count || 0
                }
            }
        }
    };
}

export function compliance(state, { payload }) {
    return {
        ...state,
        systemIssues: {
            ...state?.systemIssues,
            compliance: {
                isLoaded: true,
                profiles: payload?.data?.system?.profiles
            }
        }
    };
}

export function isPending(state, type) {
    return {
        ...state,
        systemIssues: {
            ...state?.systemIssues,
            ...type && { [type]: { isLoaded: false } }
        }
    };
}

export default {
    [SYSTEM_ISSUE_TYPES.LOAD_ADVISOR_RECOMMENDATIONS_PENDING]: (state) => isPending(state, 'advisor'),
    [SYSTEM_ISSUE_TYPES.LOAD_APPLICABLE_CVES_PENDING]: (state) => isPending(state, 'cve'),
    [SYSTEM_ISSUE_TYPES.LOAD_APPLICABLE_ADVISORIES_PENDING]: (state) => isPending(state, 'patch'),
    [SYSTEM_ISSUE_TYPES.LOAD_COMPLIANCE_POLICIES_PENDING]: (state) => isPending(state, 'compliance'),
    [SYSTEM_ISSUE_TYPES.LOAD_ADVISOR_RECOMMENDATIONS_FULFILLED]: advisor,
    [SYSTEM_ISSUE_TYPES.LOAD_APPLICABLE_CVES_FULFILLED]: cve,
    [SYSTEM_ISSUE_TYPES.LOAD_APPLICABLE_ADVISORIES_FULFILLED]: patch,
    [SYSTEM_ISSUE_TYPES.LOAD_COMPLIANCE_POLICIES_FULFILLED]: compliance
};
