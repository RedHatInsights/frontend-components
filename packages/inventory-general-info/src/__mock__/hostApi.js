import { hosts } from '../api';
import MockAdapter from 'axios-mock-adapter';

export const mock = new MockAdapter(hosts.axios);
