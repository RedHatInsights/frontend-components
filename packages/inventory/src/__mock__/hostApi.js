import { hosts, tags } from '../api';
import MockAdapter from 'axios-mock-adapter';

export const mock = new MockAdapter(hosts.axios);
export const mockTags = new MockAdapter(tags.axios);
