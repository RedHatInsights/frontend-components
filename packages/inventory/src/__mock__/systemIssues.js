import { systemIssuesInstance } from '../api';
import MockAdapter from 'axios-mock-adapter';

export const mock = new MockAdapter(systemIssuesInstance);
