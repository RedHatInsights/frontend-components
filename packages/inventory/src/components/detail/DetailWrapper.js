import React from 'react';
import {
    Drawer,
    DrawerPanelContent,
    DrawerContent,
    DrawerContentBody,
    DrawerPanelBody,
    DrawerActions,
    DrawerHead,
    DrawerCloseButton
} from '@patternfly/react-core/dist/js/components/Drawer';
import PropTypes from 'prop-types';
import { useSelector, useDispatch, useStore } from 'react-redux';
import { toggleDrawer } from '../../redux/actions';
import { BasicInfo, SystemIssues } from '../drawer';
import { Stack, StackItem } from '@patternfly/react-core/dist/js/layouts/Stack';
import FactsInfo from './FactsInfo';

const DetailWrapper = ({ children, hideInvLink, showTags, Wrapper, className, appName, ...props }) => {
    const dispatch = useDispatch();
    const store = useStore();
    const isExpanded = useSelector(({ entityDetails: { isToggleOpened } }) => isToggleOpened);
    const entity = useSelector(({ entityDetails: { entity } }) => entity);
    const loaded = useSelector(({ entityDetails: { loaded } }) => loaded);

    return <Drawer
        className={`ins-c-inventory__drawer ${className || ''}`}
        isExpanded={isExpanded}
        onExpand={() => dispatch(toggleDrawer(true))}
        {...props}
    >
        <DrawerContent
            panelContent={
                <DrawerPanelContent>
                    <DrawerHead>
                        <BasicInfo hideInvLink={ hideInvLink } showTags={ showTags } />
                        <DrawerActions>
                            <DrawerCloseButton onClick={() =>  dispatch(toggleDrawer(false))} />
                        </DrawerActions>
                    </DrawerHead>
                    <DrawerPanelBody>
                        <Stack className="ins-c-inventory__drawer--content">
                            <StackItem>
                                <SystemIssues isOpened={isExpanded} />
                            </StackItem>
                            <StackItem isFilled className="ins-c-inventory__drawer--facts">
                                <FactsInfo entity={entity} loaded={loaded} />
                                {
                                    isExpanded &&
                                    loaded &&
                                    Wrapper &&
                                    <Wrapper
                                        store={store}
                                        appName={appName}
                                    />
                                }
                            </StackItem>
                        </Stack>
                    </DrawerPanelBody>
                </DrawerPanelContent>
            }
        >
            <DrawerContentBody>
                {children}
            </DrawerContentBody>
        </DrawerContent>
    </Drawer>;
};

DetailWrapper.propTypes = {
    children: PropTypes.any,
    hideInvLink: PropTypes.bool,
    showTags: PropTypes.bool,
    appName: PropTypes.oneOf([
        'general_information',
        'advisor',
        'insights',
        'compliance',
        'vulnerabilities',
        'patch'
    ])
};

DetailWrapper.defaultProps = {
    appName: 'general_information'
};

export default DetailWrapper;
