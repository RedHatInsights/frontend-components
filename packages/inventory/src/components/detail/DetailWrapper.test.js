/* eslint-disable camelcase */
import React from 'react';
import DetailWrapper from './DetailWrapper';
import { mount } from 'enzyme';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import promiseMiddleware from 'redux-promise-middleware';
import toJson from 'enzyme-to-json';
import { mock } from '../../__mock__/systemIssues';

describe('DetailWrapper', () => {
    mock.onGet('/api/patch/v1/systems/test-id').reply(200, 'test');
    mock.onGet('/api/insights/v1/system/test-id/reports/').reply(200, 'test');
    mock.onGet('/api/vulnerability//v1/systems/test-id/cves?page=1&page_size=1&impact=2').reply(200, 'low-test');
    mock.onGet('/api/vulnerability//v1/systems/test-id/cves?page=1&page_size=1&impact=4').reply(200, 'moderate-test');
    mock.onGet('/api/vulnerability//v1/systems/test-id/cves?page=1&page_size=1&impact=5').reply(200, 'important-test');
    mock.onGet('/api/vulnerability//v1/systems/test-id/cves?page=1&page_size=1&impact=7').reply(200, 'critical-test');
    mock.onGet('/api/vulnerability//v1/systems/test-id/cves?page=1&page_size=1&impact=2').reply(500);
    mock.onPost('/api/compliance/graphql').reply(200, 'test');
    let initialState;
    let mockStore;
    const _Date = Date;
    const currDate = new Date('1970');
    beforeAll(() => {
        /*eslint no-global-assign:off*/
        Date = class extends Date {
            constructor(...props) {
                if (props.length > 0) {
                    return new _Date(...props);
                }

                return currDate;
            }

            static now() {
                return new _Date('1970').getTime();
            }
        };
    });

    afterAll(() => {
        Date = _Date;
    });
    beforeEach(() => {
        mockStore = configureStore([ promiseMiddleware() ]);
        initialState = {
            entityDetails: {
                loaded: true,
                entity: {
                    id: 'some-id',
                    updated: new Date(),
                    culled_timestamp: new Date(),
                    stale_warning_timestamp: new Date(),
                    stale_timestamp: new Date()
                },
                isToggleOpened: true
            }
        };
    });

    describe('DOM', () => {
        it('should render without data', () => {
            const store = mockStore({
                entityDetails: {}
            });
            const wrapper = mount(<Provider store={ store }>
                <DetailWrapper/>
            </Provider>);
            expect(toJson(wrapper.find('DetailWrapper'), { mode: 'deep' })).toMatchSnapshot();
        });

        it('should render with data', () => {
            const store = mockStore(initialState);
            const wrapper = mount(<Provider store={ store }>
                <DetailWrapper/>
            </Provider>);
            expect(toJson(wrapper.find('DetailWrapper'), { mode: 'deep' })).toMatchSnapshot();
        });

        it('should pass correct props to BasicInfo', () => {
            const store = mockStore(initialState);
            const wrapper = mount(<Provider store={ store }>
                <DetailWrapper/>
            </Provider>);
            expect(toJson(wrapper.find('BasicInfo'), { mode: 'shallow' })).toMatchSnapshot();
        });

        it('should render Wrapper', () => {
            const store = mockStore(initialState);
            const wrapper = mount(<Provider store={ store }>
                <DetailWrapper Wrapper={() => <div className="test">something</div>}/>
            </Provider>);
            expect(toJson(wrapper.find('.ins-c-inventory__drawer--facts'), { mode: 'deep' })).toMatchSnapshot();
        });

        it('should render children', () => {
            const store = mockStore(initialState);
            const wrapper = mount(<Provider store={ store }>
                <DetailWrapper>
                    <div className="test">something</div>
                </DetailWrapper>
            </Provider>);
            expect(toJson(wrapper.find('.pf-c-drawer__body .test'), { mode: 'shallow' })).toMatchSnapshot();
        });

        it('should calculate classnames', () => {
            const store = mockStore(initialState);
            const wrapper = mount(<Provider store={ store }>
                <DetailWrapper className="test"/>
            </Provider>);
            expect(wrapper.find('.ins-c-inventory__drawer.test')).toBeDefined();
        });
    });

    describe('API', () => {
        it('should call open', () => {
            const store = mockStore(initialState);
            const wrapper = mount(<Provider store={ store }>
                <DetailWrapper className="test"/>
            </Provider>);
            wrapper.find('.pf-c-drawer__close button').simulate('click');
            const actions = store.getActions();
            expect(actions[actions.length - 1]).toMatchObject({ payload: { isOpened: false }, type: 'TOGGLE_INVENTORY_DRAWER' });
        });
    });
});
