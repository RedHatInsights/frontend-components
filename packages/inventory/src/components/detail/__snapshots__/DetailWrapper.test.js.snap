// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetailWrapper DOM should pass correct props to BasicInfo 1`] = `
<BasicInfo
  hideInvLink={false}
  showTags={false}
/>
`;

exports[`DetailWrapper DOM should render Wrapper 1`] = `
Array [
  <div
    className="pf-l-stack__item pf-m-fill ins-c-inventory__drawer--facts"
  >
    <div
      className="pf-l-grid ins-entity-facts"
    >
      <div
        className="pf-l-grid__item pf-m-6-col-on-md"
      >
        <div>
          <span>
            UUID:
          </span>
          <span>
            some-id
          </span>
        </div>
        <div>
          <span>
            Last seen:
          </span>
          <span>
            Array [
              <span
                className="ins-c-inventory__culling-danger"
              >
                <svg
                  aria-hidden={true}
                  aria-labelledby={null}
                  fill="currentColor"
                  height="1em"
                  role="img"
                  style={
                    Object {
                      "verticalAlign": "-0.125em",
                    }
                  }
                  viewBox="0 0 512 512"
                  width="1em"
                >
                  <path
                    d="M504 256c0 136.997-111.043 248-248 248S8 392.997 8 256C8 119.083 119.043 8 256 8s248 111.083 248 248zm-248 50c-25.405 0-46 20.595-46 46s20.595 46 46 46 46-20.595 46-46-20.595-46-46-46zm-43.673-165.346l7.418 136c.347 6.364 5.609 11.346 11.982 11.346h48.546c6.373 0 11.635-4.982 11.982-11.346l7.418-136c.375-6.874-5.098-12.654-11.982-12.654h-63.383c-6.884 0-12.356 5.78-11.981 12.654z"
                    transform=""
                  />
                </svg>
                01 Jan 1970 00:00 UTC
              </span>,
              <Portal
                containerInfo={
                  <div>
                    <div
                      class=""
                      id=""
                      role="tooltip"
                    >
                      <div
                        class="pf-c-tooltip__content"
                      >
                        System scheduled for inventory removal in 0 days
                      </div>
                    </div>
                  </div>
                }
              >
                <div
                  className=""
                  id=""
                  role="tooltip"
                >
                  <div
                    className="pf-c-tooltip__content"
                  >
                    System scheduled for inventory removal in 0 days
                  </div>
                </div>
              </Portal>,
            ]
          </span>
        </div>
      </div>
    </div>
    <div
      className="test"
    >
      something
    </div>
  </div>,
  <div
    className="pf-l-stack__item pf-m-fill ins-c-inventory__drawer--facts"
  >
    <div
      className="pf-l-grid ins-entity-facts"
    >
      <div
        className="pf-l-grid__item pf-m-6-col-on-md"
      >
        <div>
          <span>
            UUID:
          </span>
          <span>
            some-id
          </span>
        </div>
        <div>
          <span>
            Last seen:
          </span>
          <span>
            Array [
              <span
                className="ins-c-inventory__culling-danger"
              >
                <svg
                  aria-hidden={true}
                  aria-labelledby={null}
                  fill="currentColor"
                  height="1em"
                  role="img"
                  style={
                    Object {
                      "verticalAlign": "-0.125em",
                    }
                  }
                  viewBox="0 0 512 512"
                  width="1em"
                >
                  <path
                    d="M504 256c0 136.997-111.043 248-248 248S8 392.997 8 256C8 119.083 119.043 8 256 8s248 111.083 248 248zm-248 50c-25.405 0-46 20.595-46 46s20.595 46 46 46 46-20.595 46-46-20.595-46-46-46zm-43.673-165.346l7.418 136c.347 6.364 5.609 11.346 11.982 11.346h48.546c6.373 0 11.635-4.982 11.982-11.346l7.418-136c.375-6.874-5.098-12.654-11.982-12.654h-63.383c-6.884 0-12.356 5.78-11.981 12.654z"
                    transform=""
                  />
                </svg>
                01 Jan 1970 00:00 UTC
              </span>,
              <Portal
                containerInfo={
                  <div>
                    <div
                      class=""
                      id=""
                      role="tooltip"
                    >
                      <div
                        class="pf-c-tooltip__content"
                      >
                        System scheduled for inventory removal in 0 days
                      </div>
                    </div>
                  </div>
                }
              >
                <div
                  className=""
                  id=""
                  role="tooltip"
                >
                  <div
                    className="pf-c-tooltip__content"
                  >
                    System scheduled for inventory removal in 0 days
                  </div>
                </div>
              </Portal>,
            ]
          </span>
        </div>
      </div>
    </div>
    <div
      className="test"
    >
      something
    </div>
  </div>,
]
`;

exports[`DetailWrapper DOM should render children 1`] = `
<div
  className="test"
>
  something
</div>
`;

exports[`DetailWrapper DOM should render with data 1`] = `
<div
  className="pf-c-drawer pf-m-expanded ins-c-inventory__drawer "
  onExpand={[Function]}
>
  <div
    className="pf-c-drawer__main"
  >
    <div
      className="pf-c-drawer__content"
    >
      <div
        className="pf-c-drawer__body"
      />
    </div>
    <div
      className="pf-c-drawer__panel"
      hidden={false}
    >
      <div
        className="pf-c-drawer__body"
      >
        <div
          className="pf-c-drawer__head"
        >
          <div
            className="pf-l-stack pf-m-gutter"
          >
            <div
              className="pf-l-stack__item"
            >
              <div
                className="pf-l-split"
              >
                <div
                  className="pf-l-split__item pf-m-fill"
                >
                  <h3
                    className="pf-c-title pf-m-xl"
                  />
                </div>
                <div
                  className="pf-l-split__item"
                >
                  <a
                    className="ins-c-entity-detail__inv-link"
                    href="./insights/inventory/some-id"
                  >
                    Open in Inventory
                  </a>
                </div>
              </div>
            </div>
          </div>
          <div
            className="pf-c-drawer__actions"
          >
            <div
              className="pf-c-drawer__close"
              onClick={[Function]}
            >
              <button
                aria-disabled={null}
                aria-label="Close drawer panel"
                className="pf-c-button pf-m-plain"
                data-ouia-component-id={null}
                data-ouia-component-type="PF4/Button"
                data-ouia-safe={true}
                disabled={false}
                onClick={[Function]}
                tabIndex={null}
                type="button"
              >
                <svg
                  aria-hidden={true}
                  aria-labelledby={null}
                  fill="currentColor"
                  height="1em"
                  role="img"
                  style={
                    Object {
                      "verticalAlign": "-0.125em",
                    }
                  }
                  viewBox="0 0 352 512"
                  width="1em"
                >
                  <path
                    d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
                    transform=""
                  />
                </svg>
              </button>
            </div>
          </div>
        </div>
      </div>
      <div
        className="pf-c-drawer__body"
      >
        <div
          className="pf-l-stack ins-c-inventory__drawer--content"
        >
          <div
            className="pf-l-stack__item"
          >
            <div
              className="pf-l-stack pf-m-gutter ins-c-inventory__drawer-system-issues"
            >
              <div
                className="pf-l-stack__item"
              >
                <h4
                  className="pf-c-title pf-m-lg"
                >
                  Top system issues
                </h4>
              </div>
              <div
                className="pf-l-stack__item"
              >
                <div
                  className="pf-l-level"
                >
                  <div>
                    <div
                      className="pf-l-stack pf-m-gutter"
                    >
                      <div
                        className="pf-l-stack__item"
                      >
                        Applicable CVEs
                      </div>
                      <div
                        className="pf-l-stack__item"
                      >
                        <div
                          className="ins-c-skeleton ins-c-skeleton__md"
                        >
                          Â 
                        </div>
                      </div>
                    </div>
                  </div>
                  <div>
                    <div
                      className="pf-l-stack pf-m-gutter"
                    >
                      <div
                        className="pf-l-stack__item"
                      >
                        Applicable advisories
                      </div>
                      <div
                        className="pf-l-stack__item"
                      >
                        <div
                          className="ins-c-skeleton ins-c-skeleton__md"
                        >
                          Â 
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              <div
                className="pf-l-stack__item"
              >
                <div
                  className="ins-c-skeleton ins-c-skeleton__md"
                >
                  Â 
                </div>
              </div>
              <div
                className="ins-c-skeleton ins-c-skeleton__md"
              >
                Â 
              </div>
            </div>
          </div>
          <div
            className="pf-l-stack__item pf-m-fill ins-c-inventory__drawer--facts"
          >
            <div
              className="pf-l-grid ins-entity-facts"
            >
              <div
                className="pf-l-grid__item pf-m-6-col-on-md"
              >
                <div>
                  <span>
                    UUID:
                  </span>
                  <span>
                    some-id
                  </span>
                </div>
                <div>
                  <span>
                    Last seen:
                  </span>
                  <span>
                    Array [
                      <span
                        className="ins-c-inventory__culling-danger"
                      >
                        <svg
                          aria-hidden={true}
                          aria-labelledby={null}
                          fill="currentColor"
                          height="1em"
                          role="img"
                          style={
                            Object {
                              "verticalAlign": "-0.125em",
                            }
                          }
                          viewBox="0 0 512 512"
                          width="1em"
                        >
                          <path
                            d="M504 256c0 136.997-111.043 248-248 248S8 392.997 8 256C8 119.083 119.043 8 256 8s248 111.083 248 248zm-248 50c-25.405 0-46 20.595-46 46s20.595 46 46 46 46-20.595 46-46-20.595-46-46-46zm-43.673-165.346l7.418 136c.347 6.364 5.609 11.346 11.982 11.346h48.546c6.373 0 11.635-4.982 11.982-11.346l7.418-136c.375-6.874-5.098-12.654-11.982-12.654h-63.383c-6.884 0-12.356 5.78-11.981 12.654z"
                            transform=""
                          />
                        </svg>
                        01 Jan 1970 00:00 UTC
                      </span>,
                      <Portal
                        containerInfo={
                          <div>
                            <div
                              class=""
                              id=""
                              role="tooltip"
                            >
                              <div
                                class="pf-c-tooltip__content"
                              >
                                System scheduled for inventory removal in 0 days
                              </div>
                            </div>
                          </div>
                        }
                      >
                        <div
                          className=""
                          id=""
                          role="tooltip"
                        >
                          <div
                            className="pf-c-tooltip__content"
                          >
                            System scheduled for inventory removal in 0 days
                          </div>
                        </div>
                      </Portal>,
                    ]
                  </span>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`DetailWrapper DOM should render without data 1`] = `
<div
  className="pf-c-drawer ins-c-inventory__drawer "
  onExpand={[Function]}
>
  <div
    className="pf-c-drawer__main"
  >
    <div
      className="pf-c-drawer__content"
    >
      <div
        className="pf-c-drawer__body"
      />
    </div>
    <div
      className="pf-c-drawer__panel"
      hidden={true}
    >
      <div
        className="pf-c-drawer__body"
      >
        <div
          className="pf-c-drawer__head"
        >
          <div
            className="pf-l-stack pf-m-gutter"
          >
            <div
              className="pf-l-stack__item"
            >
              <div
                className="pf-l-split"
              >
                <div
                  className="pf-l-split__item pf-m-fill"
                >
                  <h3
                    className="pf-c-title pf-m-xl"
                  />
                </div>
                <div
                  className="pf-l-split__item"
                >
                  <a
                    className="ins-c-entity-detail__inv-link"
                    href="./insights/inventory/undefined"
                  >
                    Open in Inventory
                  </a>
                </div>
              </div>
            </div>
          </div>
          <div
            className="pf-c-drawer__actions"
          >
            <div
              className="pf-c-drawer__close"
              onClick={[Function]}
            >
              <button
                aria-disabled={null}
                aria-label="Close drawer panel"
                className="pf-c-button pf-m-plain"
                data-ouia-component-id={null}
                data-ouia-component-type="PF4/Button"
                data-ouia-safe={true}
                disabled={false}
                onClick={[Function]}
                tabIndex={null}
                type="button"
              >
                <svg
                  aria-hidden={true}
                  aria-labelledby={null}
                  fill="currentColor"
                  height="1em"
                  role="img"
                  style={
                    Object {
                      "verticalAlign": "-0.125em",
                    }
                  }
                  viewBox="0 0 352 512"
                  width="1em"
                >
                  <path
                    d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
                    transform=""
                  />
                </svg>
              </button>
            </div>
          </div>
        </div>
      </div>
      <div
        className="pf-c-drawer__body"
      >
        <div
          className="pf-l-stack ins-c-inventory__drawer--content"
        >
          <div
            className="pf-l-stack__item"
          >
            <div
              className="pf-l-stack pf-m-gutter ins-c-inventory__drawer-system-issues"
            >
              <div
                className="pf-l-stack__item"
              >
                <h4
                  className="pf-c-title pf-m-lg"
                >
                  Top system issues
                </h4>
              </div>
              <div
                className="pf-l-stack__item"
              >
                <div
                  className="pf-l-level"
                >
                  <div>
                    <div
                      className="pf-l-stack pf-m-gutter"
                    >
                      <div
                        className="pf-l-stack__item"
                      >
                        Applicable CVEs
                      </div>
                      <div
                        className="pf-l-stack__item"
                      >
                        <div
                          className="ins-c-skeleton ins-c-skeleton__md"
                        >
                          Â 
                        </div>
                      </div>
                    </div>
                  </div>
                  <div>
                    <div
                      className="pf-l-stack pf-m-gutter"
                    >
                      <div
                        className="pf-l-stack__item"
                      >
                        Applicable advisories
                      </div>
                      <div
                        className="pf-l-stack__item"
                      >
                        <div
                          className="ins-c-skeleton ins-c-skeleton__md"
                        >
                          Â 
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              <div
                className="pf-l-stack__item"
              >
                <div
                  className="ins-c-skeleton ins-c-skeleton__md"
                >
                  Â 
                </div>
              </div>
              <div
                className="ins-c-skeleton ins-c-skeleton__md"
              >
                Â 
              </div>
            </div>
          </div>
          <div
            className="pf-l-stack__item pf-m-fill ins-c-inventory__drawer--facts"
          >
            <div
              className="pf-l-grid ins-entity-facts"
            >
              <div
                className="pf-l-grid__item pf-m-6-col-on-md"
              >
                <div>
                  <span>
                    UUID:
                  </span>
                  <span>
                    <div
                      className="ins-c-skeleton ins-c-skeleton__md"
                    >
                      Â 
                    </div>
                  </span>
                </div>
                <div>
                  <span>
                    Last seen:
                  </span>
                  <span>
                    <div
                      className="ins-c-skeleton ins-c-skeleton__sm"
                    >
                      Â 
                    </div>
                  </span>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
